pkg/operator/controller/externaldns/pod.go

```diff
@@ -164,8 +165,8 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 		args = append(args, "--ignore-hostname-annotation")
 	}
 
-	if len(b.externalDNS.Spec.Source.FQDNTemplate) > 0 {
-		args = append(args, fmt.Sprintf("--fqdn-template=%s", b.externalDNS.Spec.Source.FQDNTemplate))
+	if b.externalDNS.Spec.Source.FQDNTemplate != nil && len(b.externalDNS.Spec.Source.FQDNTemplate) > 0 {
```

Comment:
Checking for `nil` is superfluous.  `len(b.externalDNS.Spec.Source.FQDNTemplate) > 0` is sufficient.